microcosm_fastapi/exception_handler.py

```diff
@@ -0,0 +1,28 @@
+from fastapi import Request
+from fastapi.responses import JSONResponse
+
+from microcosm_fastapi.errors import ParsedException
+from microcosm_fastapi.utils import bind_to_request_state
+import traceback
+
+
+async def global_exception_handler(request: Request, call_next):
+    """
+    Catches exceptions and converts them into JSON responses that can be returned back to the client
+    to fit in with existing microcosm conventions
+
+    """
+    try:
+        return await call_next(request)
+    except Exception as error:
+        bind_to_request_state(request, error=error, traceback=traceback.format_exc(limit=10))
+        parsed_exception = ParsedException(error)
+        return JSONResponse(status_code=parsed_exception.status_code, content=parsed_exception.to_dict())
```

Comment:
not for this PR, but I guess this is where we'd put a Sentry integration?

Reply:
yep exactly